### PR TITLE
grc: Converted actions to Gio.Action instead of Gtk.Action

### PR DESCRIPTION
**Pull request by sdh11**
The Gtk.Action class in Gtk3 has been deprecated in favor of Gio.Action. This converts all of the actions and rebuilds the menus/toolbar structures to handle the newer style.

### CODE REVIEW

**Comment by jmcorgan** [comment]
@skoslowski Do you mind if I go ahead and merge this into python3 so we can just evaluate it there?  Right now it is a fast forward.

**Comment by skoslowski** [comment]
I will review over the weekend. I am sure Seth is happy to rebase it, if
something else changes until then

File: grc/gui/DrawingArea.py [hunk 1]
```diff
@@ -155,6 +164,41 @@ def _handle_mouse_motion(self, widget, event):
...
+            Gtk.accelerator_parse('Add'): Actions.PORT_CONTROLLER_INC,
+            Gtk.accelerator_parse('Subtract'): Actions.PORT_CONTROLLER_DEC,
+            Gtk.accelerator_parse('Return'): Actions.BLOCK_PARAM_MODIFY,
+            Gtk.accelerator_parse('<Shift>t'): Actions.BLOCK_VALIGN_TOP,
+            Gtk.accelerator_parse('<Shift>m'): Actions.BLOCK_VALIGN_MIDDLE,
+            Gtk.accelerator_parse('<Shift>b'): Actions.BLOCK_VALIGN_BOTTOM,
+            Gtk.accelerator_parse('<Shift>l'): Actions.BLOCK_HALIGN_LEFT,
+            Gtk.accelerator_parse('<Shift>c'): Actions.BLOCK_HALIGN_CENTER,
+            Gtk.accelerator_parse('<Shift>r'): Actions.BLOCK_HALIGN_RIGHT,
+        }
```
Comment (skoslowski) [inline grc/gui/DrawingArea.py]: this should probably be a module variable (not recreated for each key-press event)

File: grc/main.py [hunk 2]
```diff
@@ -65,16 +62,23 @@ def main():
...
 
     #formatter = utils.log.ConsoleFormatter()
     console.setFormatter(formatter)
     log.addHandler(console)
 
-    log.debug("Loading platform")
+    log.debug("Running main")
 
+    # Delay importing until the logging is setup
+    # Otherwise, the decorators could not use logging.
```
Comment (skoslowski) [inline grc/main.py]: which decorators?

File: grc/gui/Notebook.py [hunk 3]
```diff
@@ -99,20 +129,12 @@ def __init__(self, main_window, flow_graph, file_path=''):
...
 
-    def _handle_scroll_window_key_press(self, widget, event):
-        """forward Ctrl-PgUp/Down to NotebookPage (switch fg instead of horiz. scroll"""
-        is_ctrl_pg = (
-            event.state & Gdk.ModifierType.CONTROL_MASK and
-            event.keyval in (Gdk.KEY_Page_Up, Gdk.KEY_Page_Down)
-        )
-        if is_ctrl_pg:
-            return self.get_parent().event(event)
+
```
Comment (skoslowski) [inline grc/gui/Notebook.py]: is this no longer required?

File: grc/gui/MainWindow.py [hunk 4]
```diff
@@ -58,18 +59,29 @@ def __init__(self, app, platform, action_handler_callback):
         log.debug("__init__()")
 
         self._platform = platform
+        self.app = app
         self.config = platform.config
 
+        # Add all "win" actions to the local
+        win_actions = filter(lambda x: x.startswith("win."), Actions.get_actions())
+        map(lambda x: self.add_action(Actions.actions[x]), win_actions)
```
Comment (skoslowski) [inline grc/gui/MainWindow.py]: style: lambda with side effect only is better expressed as a for loop ... with if clause

File: grc/gui/DrawingArea.py [hunk 5]
```diff
@@ -155,6 +164,41 @@ def _handle_mouse_motion(self, widget, event):
...
+        # Not sold on this.
+        if (key, mod) in accels:
+            accels[(key, mod)]()
+            return True
```
Comment (skoslowski) [inline grc/gui/DrawingArea.py]: maybe some like
```
try:
    action = accels[(key, mod)]
except KeyError:
    return
else:
    action()
```
if there is no gtk way to do the binding this is okay...

File: grc/gui/MainWindow.py (hunk 4 above)
Comment (noc0lour) [inline grc/gui/MainWindow.py]: map and lambda, Guido van Rossum would hate you (: 
@skoslowski wouldn't a list comprehension work as well, or is that what you meant?

Comment (skoslowski) [inline grc/gui/MainWindow.py]: map and lambda have valid use cases.

As no list is build here, a list comprehension isn't much better here IMO.

in python3 map() is a generator. I'm not really if this even had the intended function

**Comment by jmcorgan** [comment]
I'm going to squash this branch into a single commit, to make the history cleaner, then put that on the tip of the current python3 branch. @sdh11 this will mean that you'll need to regenerate your local development branch on top of this.

File: grc/gui/MainWindow.py (hunk 4 above)
Comment (sdh11) [inline grc/gui/MainWindow.py]: Nope, it doesn't have the intended function; don't know how I missed that.

File: grc/gui/Notebook.py (hunk 3 above)
Comment (sdh11) [inline grc/gui/Notebook.py]: I'm not sure why I deleted that. I'll put it back.

File: grc/gui/DrawingArea.py (hunk 5 above)
Comment (sdh11) [inline grc/gui/DrawingArea.py]: At the moment, I think this is the best way to do it.

**Pull request closed without merging**